www/md_base/src/app/common/common.constant.coffee

```diff
@@ -5,17 +5,27 @@ invert_constant = (constant_name, inverted_constant_name) ->
 
 class Baseurlapi extends Constant('common')
     constructor: ->
-        return 'api/v2/'
```

Comment:
But some resources such as HTML template seems won't easily served without
python server. hmmm...
Give me some time to study on this.

On Wed, May 6, 2015 at 8:22 PM Pierre Tardy <notifications@github.com>
wrote:

> In www/md_base/src/app/common/common.constant.coffee
> <https://github.com/buildbot/buildbot/pull/1654#discussion_r29752536>:
>
> > @@ -5,17 +5,27 @@ invert_constant = (constant_name, inverted_constant_name) ->
> >
> >  class Baseurlapi extends Constant('common')
> >      constructor: ->
> > -        return 'api/v2/'
>
> I would prefer you can make a better proxy solution that would not involve
> such hack.
>
> With proxy, you are supposed to serve all the static frontend thing
> directly, and only proxy api/ sse/ and ws/
>
> So your workflow is supposed to use http://localhost:8021/ as the test
> server, and you don't lunch any python server
>
> This is problematic with the detection of plugins.
> We have chat with andras about that yesterday evening, please synchronise.
>
> We should find a long term solution for this problem. This is not only for
> development of this feature. For all frontend development, proxy is a very
> usefull feature. It has to be resolved with backend solution only not
> front-end hacks. The backend solution might eventually involve launching a
> python script that will report a json file with the plugin configuration
> available.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1654/files#r29752536>.
>